Build process is broken in mouse.cpp on commit d23038f

### Describe the bug

When I try to build on my Archlinux laptop source code from commit https://github.com/joncampbell123/dosbox-x/commit/d23038fb7be7225f19890bf11097e3b3a965e645 I got an error in mouse.cpp:

```
mouse.cpp: In function 'void DrawCursorText()':
mouse.cpp:549:34: error: 'Bit8u' was not declared in this scope; did you mean 'Bitu'?
  549 |             mouse.backData[2] = (Bit8u)(result2 & 0xFF);
      |                                  ^~~~~
      |                                  Bitu
mouse.cpp:556:34: error: 'Bit8u' was not declared in this scope; did you mean 'Bitu'?
  556 |             mouse.backData[2] = (Bit8u)(result2 & 0xFF);
      |                                  ^~~~~
      |                                  Bitu
mouse.cpp: In function 'void Mouse_AfterNewVideoMode(bool)':
mouse.cpp:1503:29: error: 'Bit16s' was not declared in this scope; did you mean 'Bits'?
 1503 |             mouse.gran_x = (Bit16s)0xfff8;
      |                             ^~~~~~
      |                             Bits
int_dosv.cpp: In function 'bool readPCF(FILE*, int)':
int_dosv.cpp:470:85: warning: comparison is always true due to limited range of data type [-Wtype-limits]
  470 |     if ((m.height() == height || m.height() == 15) && charcode >= 0x100 && charcode <= 0xffff) {
      |                                                                            ~~~~~~~~~^~~~~~~~~
int_dosv.cpp:478:26: warning: comparison of integer expressions of different signedness: 'size_t' {aka 'long unsigned int'} and 'int' [-Wsign-compare]
  478 |         if (j != 30 && j != height * 2) continue;
      |                        ~~^~~~~~~~~~~~~
int_dosv.cpp: In function 'Bitu mskanji_api()':
int_dosv.cpp:1840:24: warning: variable 'mode' set but not used [-Wunused-but-set-variable]
 1840 |         uint16_t func, mode;
      |                        ^~~~
int_dosv.cpp: In function 'void DOSV_CursorXor24(Bitu, Bitu, Bitu, Bitu)':
int_dosv.cpp:1935:26: warning: variable 'dummy' set but not used [-Wunused-but-set-variable]
 1935 |         volatile uint8_t dummy;
      |                          ^~~~~
int_dosv.cpp: In function 'void DOSV_CursorXor(Bitu, Bitu)':
int_dosv.cpp:2013:34: warning: variable 'dummy' set but not used [-Wunused-but-set-variable]
 2013 |                 volatile uint8_t dummy;
      |                                  ^~~~~
make[3]: *** [Makefile:412: mouse.o] Error 1
```

Looks like last changes on mouse.cpp break building process.


### Steps to reproduce the behaviour

1. Grab latest source code
2. Try to get it build
3. Wait a little

### Expected behavior

Building process going on.

### What operating system(s) this bug have occurred on?

Archlinux

### What version(s) of DOSBox-X have this bug?

Commit d23038f

### Used configuration

```ini

```

### Output log

```text

```

### Additional information

_No response_

### Have you checked that no similar bug report(s) exist?

- [x] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [x] I agree to follow the code of conduct and the contributing guidelines.

--- a/src/ints/mouse.cpp
+++ b/src/ints/mouse.cpp
@@ -546,15 +546,15 @@ void DrawCursorText() {
         if(mouse.backData[4] == 1) {
             uint16_t result2;
             ReadCharAttr(mouse.backposx + 1, mouse.backposy, page, &result2);
-            mouse.backData[2] = (Bit8u)(result2 & 0xFF);
-            mouse.backData[3] = (Bit8u)(result2>>8);
+            mouse.backData[2] = (uint8_t)(result2 & 0xFF);
+            mouse.backData[3] = (uint8_t)(result2>>8);
             WriteChar((uint16_t)mouse.backposx,mouse.backposy,page,(uint8_t)(result&0xFF),(uint8_t)(result>>8),true);
             WriteChar((uint16_t)(mouse.backposx+1),mouse.backposy,page,(uint8_t)(result2&0xFF),(uint8_t)(result>>8),true);
         } else if(mouse.backData[4] == 2) {
             uint16_t result2;
             ReadCharAttr(mouse.backposx - 1, mouse.backposy, page, &result2);
-            mouse.backData[2] = (Bit8u)(result2 & 0xFF);
-            mouse.backData[3] = (Bit8u)(result2>>8);
+            mouse.backData[2] = (uint8_t)(result2 & 0xFF);
+            mouse.backData[3] = (uint8_t)(result2>>8);
             WriteChar((uint16_t)(mouse.backposx-1),(uint16_t)mouse.backposy,page,(uint8_t)(result2&0xFF),(uint8_t)(result>>8),true);
             WriteChar((uint16_t)mouse.backposx,(uint16_t)mouse.backposy,page,(uint8_t)(result&0xFF),(uint8_t)(result>>8),true);
         } else {
@@ -1500,8 +1500,8 @@ void Mouse_AfterNewVideoMode(bool setmode) {
         break;
     case 0x70:
         if(IS_DOSV && DOSV_CheckCJKVideoMode()) {
-            mouse.gran_x = (Bit16s)0xfff8;
-            mouse.gran_y = (Bit16s)0xfff8;
+            mouse.gran_x = (int16_t)0xfff8;
+            mouse.gran_y = (int16_t)0xfff8;
             mouse.max_y = (real_readb(BIOSMEM_SEG, BIOSMEM_NB_ROWS) + 1) * 8 - 1;
             mouse.max_x = real_readw(BIOSMEM_SEG, BIOSMEM_NB_COLS) * 8 - 1;
             break;

